mailpoet/lib/AdminPages/Pages/AutomationEditor.php

```diff
@@ -0,0 +1,69 @@
+<?php declare(strict_types=1);
+
+namespace MailPoet\AdminPages\Pages;
+
+use DateTimeImmutable;
+use MailPoet\AdminPages\PageRenderer;
+use MailPoet\Automation\Engine\Storage\WorkflowStorage;
+use MailPoet\Automation\Engine\Workflows\Step;
+use MailPoet\Automation\Engine\Workflows\Workflow;
+use MailPoet\WP\Functions as WPFunctions;
+
+class AutomationEditor {
+  /** @var WorkflowStorage */
+  private $workflowStorage;
+
+  /** @var PageRenderer */
+  private $pageRenderer;
+
+  /** @var WPFunctions */
+  private $wp;
+
+  public function __construct(
+    WorkflowStorage $workflowStorage,
+    PageRenderer $pageRenderer,
+    WPFunctions $wp
+  ) {
+    $this->workflowStorage = $workflowStorage;
+    $this->pageRenderer = $pageRenderer;
+    $this->wp = $wp;
+  }
+
+  public function render() {
+    // Gutenberg styles
+    $this->wp->wpEnqueueStyle('wp-edit-post');
+    $this->wp->wpEnqueueStyle('wp-format-library');
+    $this->wp->wpEnqueueMedia();
+
+    $id = isset($_GET['id']) ? (int)$_GET['id'] : null;
+
+    $workflow = $id ? $this->workflowStorage->getWorkflow($id) : null;
+    $this->pageRenderer->displayPage('automation/editor.html', [
+      'workflow' => $workflow ? $this->buildWorkflow($workflow) : null,
```

Comment:
I assume a fallback when no workflow was found would be handled at a later stage.

Reply:
Yeah, noting it.